### PR TITLE
feat: (Core) add semantic icons to Message box and remove the close button

### PR DESCRIPTION
**Pull request by InnaAtanasova**
#### Please provide a link to the associated issue.
Closes https://github.com/SAP/fundamental-ngx/issues/3887

#### Please provide a brief summary of this pull request.
- introduced semantic icons into Message box component
- removed the close button as per Fiori specs
- the Switch component is broken but it comes from v.0.14.0 of Fundamental-styles and is fixed in [this PR](https://github.com/SAP/fundamental-ngx/pull/4116) 

BREAKING CHANGE:
`MessageBoxCloseIconComponent` component is removed

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Comment by InnaAtanasova** [comment]
**NOTE:** tests are failing for FlexibleColumnLayout but they are fixed in 2 other PRs (just need someone to review them :) )
https://github.com/SAP/fundamental-ngx/pull/4108/files and https://github.com/SAP/fundamental-ngx/pull/4116

As mentioned in the template, Switch component is broken but it's due to v.0.14.0 which is fixed in https://github.com/SAP/fundamental-ngx/pull/4116

**Review by IlyaSurmay — APPROVED** [review]

File: libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+/**
+ * Message box element representing the semantic icon in the message box header.
+ *
+ * ```html
+ * <fd-message-box-semantic-icon></fd-message-box-semantic-icon>
+ * ```
+ */
+@Component({
+    selector: 'fd-message-box-semantic-icon',
+    template: `<i [ngClass]="'sap-icon--' + this._semanticIcon" role="presentation"></i>`,
```
Comment (salarenko) [inline libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts:17]: no `this.` required

**Review by salarenko — CHANGES_REQUESTED** [review]
1. RIP close button. Do we have to remove this feature? It looks like it also should't be supported in the Dialog component, but we keep it there. It's weird that the design excluded such common element.
2. Should we force user to put the icon element in the header by himself? Maybe we could put it always in the default message-box header template and show the user how to compose it by himself in the " Complex template" example?

File: libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts (hunk 1 above)
Comment (InnaAtanasova) [reply libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts:17]: 1. It's not up to us to decide if a button should be show or not. There's no button in the designs and in the UX guidelines so if we want a component to be following Fiori3 we should not display such button. 
2. will check this and let you know :)

File: libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        this.messageBoxConfig = this.messageBoxConfig || {};
+    }
+
+    /** @hidden */
+    get _showSemanticIcon(): boolean {
+        return this.messageBoxConfig.showSemanticIcon;
+    }
+
+    /** @hidden */
+    get _semanticIcon(): string {
```
Comment (salarenko) [inline libs/core/src/lib/message-box/message-box-semantic-icon/message-box-semantic-icon.component.ts:39]: Minor suggestion, but I think in this form it's more readable:
```
get _getIcon(): string {
    return this.glyph || this.messageBoxConfig.customSemanticIcon || this._semanticIcon;
}

get _semanticIcon(): string {
    switch (this.messageBoxConfig.type) {
       case 'error':
           return 'message-error';
       case 'success':
           return 'message-success';
       case 'warning':
           return 'message-warning';
       case 'information':
           return 'message-information';
       case 'confirmation':
           return 'question-mark';
       default:
           return '';
       }
}
```
Original form is also fine, but it looks like we don't need default value in the switch statement.

File: libs/core/src/lib/message-box/message-box-header/message-box-header.component.html [hunk 3]
```diff
@@ -8,13 +8,10 @@
...
     <ng-template #defaultTemplate>
         <div fd-bar-left>
-            <fd-bar-element [fullWidth]="true">
-                <ng-content select="[fd-title]"></ng-content>
-            </fd-bar-element>
-        </div>
-        <div fd-bar-right>
             <fd-bar-element>
-                <ng-content select="fd-message-box-close-icon"></ng-content>
+                <ng-content select="fd-message-box-semantic-icon"></ng-content>
```
Comment (salarenko) [inline libs/core/src/lib/message-box/message-box-header/message-box-header.component.html:12]: What's the point of having this content slot?

File: libs/core/src/lib/message-box/message-box-header/message-box-header.component.ts [hunk 4]
```diff
@@ -41,6 +41,14 @@ export class MessageBoxHeaderComponent extends DialogHeaderBase implements After
         this.messageBoxConfig = this.messageBoxConfig || {};
     }
 
+    /** @hidden */
+    get _showSemanticIcon(): boolean {
+        if (this.messageBoxConfig.type) {
```
Comment (salarenko) [inline libs/core/src/lib/message-box/message-box-header/message-box-header.component.ts:46]: Can we simplify this to:
```
return this.messageBoxConfig.type && this.messageBoxConfig.showSemanticIcon || this.messageBoxConfig.customSemanticIcon
```
And get rid of `hideSemanticIcon`?

**Review by salarenko — CHANGES_REQUESTED** [review]

**Review by salarenko — APPROVED** [review]

**Review by JKMarkowski — APPROVED** [review]

**Pull request merged**